### PR TITLE
feat: add support for updated proofs api_versioning

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

**Comment by cryptonemo** [comment]
Requires: https://github.com/filecoin-project/rust-fil-proofs/pull/1362

File: src/post.rs [hunk 1]
```diff
@@ -443,7 +444,7 @@ pub fn verify_window_post(
         "invalid post type provided"
     );
     ensure!(
-        registered_post_proof_type_v1.version() == Version::V1,
+        registered_post_proof_type_v1.version() == ApiVersion::V1_0_0,
```
Comment (dignifiedquire) [inline src/post.rs:447]: shouldn't this be sth like 
```rust
ensure!(registered_post_proof_type_v1.version().major() == 1)
```

Comment (cryptonemo) [reply src/post.rs:447]: Would be nice, but not possible as-is.  It would require the enum implementing `major` for every version.

Comment (cryptonemo) [reply src/post.rs:447]: We could add something like a `as_semver()` method that could allow that.  Where we basically just parse the `fmt` output of each value

Comment (dignifiedquire) [reply src/post.rs:447]: how about adding 

```rust
fn as_semver(&self) -> semver::Version {
  match self {
    V1_0_0 => Version::new(1, 0, 0),
    // ..
  }
}
```

and then `fmt` can just call that

**Comment by cryptonemo** [comment]
Requires dep updates after the next proofs release is cut

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**